src/zenml/services/service_registry.py

```diff
@@ -0,0 +1,179 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+import json
+from typing import TYPE_CHECKING, Any, Dict, Optional, Type
+from uuid import UUID
+
+from zenml.logger import get_logger
+from zenml.services.service_type import ServiceType
+from zenml.utils.singleton import SingletonMetaClass
+
+logger = get_logger(__name__)
+
+if TYPE_CHECKING:
+    from zenml.services.service import BaseService
+
+
+class ServiceRegistry(metaclass=SingletonMetaClass):
+    """Registry of service types and service instances.
+
+    The service registry provides a central place to register service types
+    as well as the service instances.
+    """
+
+    def __init__(self) -> None:
+        self.service_types: Dict[ServiceType, Type["BaseService"]] = {}
+        self.services: Dict[UUID, "BaseService"] = {}
+
+    def register_service_type(self, cls: Type["BaseService"]) -> None:
+        """Registers a new service type.
+
+        Args:
+            cls: A BaseService subclass.
+        """
+        service_type = cls.SERVICE_TYPE
+        if service_type not in self.service_types:
+            self.service_types[service_type] = cls
+            logger.debug(
+                f"Registered service class {cls} for "
+                f"service type <{service_type}>"
```

Comment:
Some logs specifying a service type don't include `<...>`, might be nice to remove it here as well to stay consistent (I think we use backticks instead in the remaining code base)